modules/gapi/include/opencv2/gapi/gcompiled_async.hpp

```diff
@@ -0,0 +1,28 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+
+#ifndef OPENCV_GAPI_GCOMPILED_ASYNC_HPP
+#define OPENCV_GAPI_GCOMPILED_ASYNC_HPP
+
+#include <future>
+#include <exception>        //for std::exception_ptr
+#include <functional>       //for std::function
+#include "opencv2/gapi/garg.hpp"
+
+namespace cv {
+    //fwd declaration
+    class GCompiled;
+
+namespace gapi{
+namespace wip {
```

Comment:
May be it worth to choose better name for namespace? Something containing "volatile".

Reply:
No, `wip` was my proposal (communicated to Anton "verbally")